swap/src/protocol.rs

```diff
@@ -139,7 +139,7 @@ impl TryInto<AliceState> for State {
 
 #[async_trait]
 pub trait Database {
-    async fn open(path: &Path) -> Result<Self> where Self: std::marker::Sized;
+    async fn open(path: PathBuf) -> Result<Self> where Self: std::marker::Sized;
```

Comment:
Does `open` have to be part of the trait? We are doing at the very top of main where we know which DB we want to construct :)

Reply:
yep you are right, I think we can remove that